docs/install/build_from_source.md

```diff
@@ -183,53 +199,88 @@ nosetests --verbose tests/python/gpu/test_nccl.py
 **Recommendation to get the best performance out of NCCL:**
 It is recommended to set environment variable NCCL_LAUNCH_MODE to PARALLEL when using NCCL version 2.1 or newer.
 
+<hr>
+
+### Build MXNet with C++
 
-### Build MXNet with Language Packages
 * To enable C++ package, just add `USE_CPP_PACKAGE=1` when you run `make` or `cmake`.
 
+<hr>
 
 ### Usage Examples
-* `-j` runs multiple jobs against multi-core CPUs. Example using all cores on Linux:
+
+* `-j` runs multiple jobs against multi-core CPUs.
+
+For example, you can specify using all cores on Linux as follows:
 
 ```bash
-make -j$(nproc)
+cmake -j$(nproc)
 ```
 
-* Build without using OpenCV:
+
+#### Recommended for Systems with NVIDIA GPUs and Intel CPUs
+* Build MXNet with `cmake` and install with MKL DNN, GPU, and OpenCV support:
 
 ```bash
-make USE_OPENCV=0
+cmake -j USE_CUDA=1 USE_CUDA_PATH=/usr/local/cuda USE_CUDNN=1 USE_MKLDNN=1
```

Comment:
we usually give these default build options for user to use while building from source. Should we have a section describing different build options from here - https://github.com/apache/incubator-mxnet/blob/master/make/config.mk explaining what each of those do so that the user can make an informed personalized choice for himself? Provide the default options, but also add this section. Thoughts?

Reply:
It's interesting that we tend to promote using MKLDNN for better performance and in the CMAKE instructions, they imply that it is turned on by default. 
However the config.mk has it turned off.
I thought it was supplied as a submodule and it is on by default. Is the config.mk out of date?

And yes, it should be mentioned what happens when you just run `make`.
Seems like right now you get:
opencv, openmp (that doesn't work on mac out the box), atlas || apple

Looking this config.mk file actually makes things even more unclear...

@anirudhacharya Can you propose the logic that is followed and the recommended flags per general situation?

Reply:
There is a conversation here - https://lists.apache.org/thread.html/ca3c858a712a66f4b3443dfb395d84e8fdf2c86e297bcd28a3a36ebd@%3Cdev.mxnet.apache.org%3E on including mkl-dnn in mxnet default build.

I think the recommended flags seem okay for now, the way it is. As and when it is decided to include MKL-DNN in the default build, then the recommended flags can also be updated.

What I was suggesting was to have a section in the documentation that describes the flags already provided in the config.mk file( each environment variable has a comment describing what it does). Please let me know if I need to help in any way here, and I would be glad to.

Reply:
The flags are good